src/libraries/apicompat/ApiCompatBaseline.netstandard2.1.xml

```diff
@@ -853,12 +853,6 @@
     <Left>netstandard2.1/netstandard.dll</Left>
     <Right>net8.0/netstandard.dll</Right>
   </Suppression>
-  <Suppression>
-    <DiagnosticId>CP0015</DiagnosticId>
-    <Target>T:System.Runtime.CompilerServices.AsyncMethodBuilderAttribute:[T:System.AttributeUsageAttribute]</Target>
-    <Left>netstandard2.1/netstandard.dll</Left>
-    <Right>net8.0/netstandard.dll</Right>
-  </Suppression>
```

Comment:
Can you please explain why all these suppressions got removed if both `ApiCompatPreserveUnnecessarySuppressions` and `ApiCompatPermitUnnecessarySuppressions` are set to `true` in `resolveContract.targets`?

Reply:
These two properties are conditionally set to true for `IsSourceProject` projects (which apicompat.proj isn't): https://github.com/dotnet/runtime/blob/a9f3262175253dd69eea72007347cd0cda9e6af0/Directory.Build.props#L403

In general, we want to disable these new features for packable projects that also have a matching reference source project. I tried to capture that in the added comment in resolveContract.targets.